types/src/v0/impls/header.rs

```diff
@@ -1099,6 +1100,36 @@ impl BlockHeader<SeqTypes> for Header {
             )?,
         })
     }
+
+    fn auth_root(&self) -> anyhow::Result<Option<[u8; 32]>> {
+        match self {
+            Header::V1(_) | Header::V2(_) | Header::V3(_) => Ok(None),
+            Header::V4(header) => {
+                let mut reward_root_bytes = Vec::new();
+                let el_root = [0; 32];
+                let placeholder_root = [0; 32];
+                header
+                    .reward_merkle_tree_root
+                    .serialize_compressed(&mut reward_root_bytes)?;
+                // The number of reward root bytes produced by `serialize_compressed()` is 48.
+                ensure!(
+                    reward_root_bytes.len() == 48,
```

Comment:
@alxiong @imabdulbasit I'm a bit confused here. We return 32 for serialized_size in `CanonicalSerialize` for `KeccakNode`

https://github.com/EspressoSystems/espresso-network/blob/d500358b0e1a769f116e0b601289ace993b647d0/types/src/v0/sparse_mt.rs#L42